underfs/swift/src/main/java/alluxio/underfs/swift/SwiftUnderFileSystem.java

```diff
@@ -245,12 +260,12 @@ public long getSpace(String path, SpaceType type) throws IOException {
 
   @Override
   public boolean isFile(String path) throws IOException {
-    return exists(path) && !isDirectory(path);
+    String pathAsFile = stripFolderSuffixIfPresent(path);
+    return getObject(pathAsFile).exists();
   }
 
   @Override
   public String[] list(String path) throws IOException {
-    path = PathUtils.normalizePath(path, PATH_SEPARATOR);
```

Comment:
normalizePath is unnecessary as it is handled in listInternal. 

I'm repurposing the listInternal indirection to make exists more efficient (stay tuned)